src/components/GlobalFilter/filterApi.ts

```diff
@@ -6,22 +6,32 @@ export const storeFilter = (tags: FlagTagsFilter, isEnabled: boolean, navigate:
   if (isEnabled) {
     const searchParams = new URLSearchParams();
     const [, SIDs, mappedTags] = flatTags(tags, false, true);
+
     if (tags?.Workloads) {
       const currWorkloads = Object.entries(tags?.Workloads || {})?.find(([, workload]) => (workload as GroupItem).isSelected)?.[0];
       if (currWorkloads) {
         searchParams.append('workloads', currWorkloads);
       }
     }
-    searchParams.append('SIDs', SIDs);
-    searchParams.append('tags', mappedTags);
+
+    if (SIDs && SIDs.length > 0) {
+      searchParams.append('SIDs', SIDs.join(','));
+    }
+
+    if (mappedTags && mappedTags.length > 0) {
+      searchParams.append('tags', mappedTags.join(','));
+    }
+
+    const finalUrl = {
```

Comment:
Just a nit, this is not a URL but a so-called `to` argument or prop in react router.